### PR TITLE
Pwd recovery

### PR DESCRIPTION
**Pull request by T7Q**
hi! 

added the following:
- password recovery functionality (both back and front)
- my profile form and connected to the backend 
- updated error handling in forms
- styled a bit here and there

### CODE REVIEW

File: frontend/src/components/user/PasswordResetForm.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        {alert.show && (
+          <Alert severity={alert.severity} onClose={() => setAlert({ ...alert, show: false })}>
+            {alert.message}
+          </Alert>
+        )}
+        <form className={classes.form} noValidate>
+          <InputField
+            values={password}
+            label="New password"
+            autocomplete="current-password"
```
Comment (lapaset) [inline frontend/src/components/user/PasswordResetForm.js]: autocomplete="new-password"

File: frontend/src/components/user/PasswordResetForm.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            required={true}
+          />
+          <InputField
+            values={confirmPassword}
+            label="Confirm password"
+            autocomplete="current-password"
+            required={true}
```
Comment (lapaset) [inline frontend/src/components/user/PasswordResetForm.js:135]: autocomplete="new-password"

**Comment by iljaSL** [comment]
There are some merge conflicts that need to be resolved.

File: frontend/src/components/user/PasswordResetForm.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+  const handleClick = async (event) => {
+    event.preventDefault();
+    try {
+      const data = {
+        ...user,
+        password: password.value,
+        confirmPassword: confirmPassword.value,
+      };
+      console.log('data', data);
```
Comment (lapaset) [inline frontend/src/components/user/PasswordResetForm.js]: Remove console.log

File: backend/utilities/email.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    to: recipient,
+    subject,
+    html: content,
+  };
+
+  transporter.sendMail(mailOptions);
+};
+
+const sendResetEmail = (recipient, name, token, req) => {
+  const link = `${getRootUrl(req)}/recoverylink?token=${token}`;
```
Comment (lapaset) [inline backend/utilities/email.js:35]: For me this resulted in:
x-webdoc://3F684866-C1BB-4CDB-82BF-90D78ADDBD65/undefined/recoverylink?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjYwMzc4MzI0ZWMzZTk0MDAxZTdlNzI4OSIsImxhbmciOiJkZSIsImlhdCI6MTYxNDI1MTUwMn0.IHq_v-QZm-5Q8_N1fRK8WWbDzfo88Pu166cEzMCYxnI

**Review by lapaset — CHANGES_REQUESTED** [review]
A couple small issues, otherwise everything ok :)

File: backend/utilities/email.js (hunk 4 above)
Comment (T7Q) [reply backend/utilities/email.js:35]: I couldn't replicate this error, I get a correct link on my side.
Have you added FRONTEND_URL_DEV to .env and ran `make build`?

Comment (lapaset) [reply backend/utilities/email.js:35]: That was it, the url was missing. Everything works now, so I will accept the merge

**Review by lapaset — APPROVED** [review]

**Pull request merged**